### DISCUSSION TITLE
Welcome sudoSignals Discussions!

### DISCUSSION THREAD

**Discussion by raganmd** [SudoMagicCode/SudoSignals#8] [General]
Welcome to the sudoSignals discussion forum. We're excited to hear your feedback and help you test the project we've been working on. Feel free to ask a question, or let us know how we can improve the Signals experience.

**Comment by Craigson** [comment]
First and foremost, this is incredible so far!  I wanted to add some feedback based on my initial explorations:

## Best Practices

In a hypothetical situation, I have an installation that has two PCs. PC A is running a TouchDesigner Application, and a NodeJS server that handles some additional logic/communication. PC B is running a separate TouchDesigner Application.  

- Will all three of these applications be part of the same SudoSignals Installation? 
- Will I download the service monitor onto PC B and it will automatically know that the two TD instances running on separate PCs are part of the same installation?


## Updating an Installation's settings

- I can't seem to find the original settings used when creating an installation. For example, I want to adjust the `auto restart` setting, but it's not available in the installation's settings. All I have access to are: General, Alerts, and Settings
- With regards to the above point, I know this configuration can be accessed from the Desktop by selecting 'Configuration' from the tray icon. It would be helpful if this was also accessible directly from the dashboard.
- The Process ID for the installation is not readily accessible. I want to connect to my installation using the NodeJS plugin and the Process ID is not easy to find.
- When I update the installation's Configuration. It closes the current TD instance and relaunches it. Is this the expected behaviour?

## Controls

- Some sort of feedback would be helpful to know that the action was successful.  The controls are nice and seamless and works like a charm - I can see the value update if I have TD and the dashboard open side-by-side, but in a real installation scenario I may not always have access to the PC to confirm.  It feels a bit "click and hope", whereas a little toast message, or some other feedback in the UI letting me know the update was successful on the TD side would be both helpful and reassuring.


## Uptime

- the uptime doesn't appear to be accurate (unless I'm misunderstanding the metric).  I installed SudoSignals and launched my TD app ~30 minutes ago, and the uptime is reading `1 day, 4 hours, 1 minute`

## Feature requests

- Have the service monitor manage the nodejs server.  At present, starting the process automatically does not work by setting the path of the nodejs file and the program as the Node executable. I Imagine it wasn't designed for this, but it would be **amazing** if SudoSignals could help replace something like `pm2-windows-service` which is incredibly unreliable.
- This feels like a very "further down the road request", but having the ability to stream logs to the dashboard would be amazing. For example, in a node application:

```
// instead of this
console.log('something happened')

// we could use sudo signals
myPlugin.log('something happened')
```

This could perhaps help add more detail to the report, which can currently only report integer values.

**Reply by raganmd** [reply]
Thanks for this thoughtful write up! There's lots in here for us to look through and think about.
